pkgs/development/python-modules/mesa/default.nix

```diff
@@ -0,0 +1,34 @@
+{ stdenv, buildPythonPackage, fetchFromGitHub, isPy27
+, cookiecutter, networkx , pandas, tornado, tqdm
+, pytest }:
+
+buildPythonPackage rec {
+  pname = "mesa";
+  version = "0.8.6";
+
+  # According to their docs, this library is for Python 3+.
+  disabled = isPy27;
+
+  src = fetchFromGitHub {
+    repo = "mesa";
+    owner = "projectmesa";
+    rev = "v${version}";
+    sha256 = "0d8c636zhswxd91ldlmdxxlyym2fj3bk1iqmpc1jp3hg7vvc7w03";
+  };
+
+  checkInputs = [ pytest ];
```

Comment:
OK, I like this. However, this way, `pytest` also runs tests for the examples provided with the library which don't run through due to the following error:
```
_____ ERROR collecting examples/wolf_sheep/wolf_sheep/test_random_walk.py ______
ImportError while importing test module '/build/source/examples/wolf_sheep/wolf_sheep/test_random_walk.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
examples/wolf_sheep/wolf_sheep/test_random_walk.py:11: in <module>
    from random_walk import RandomWalker
E   ModuleNotFoundError: No module named 'random_walk'
```

There seems to be an issue with the path? Any suggestions?

Reply:
This might be a case of the “`pytest` vs. `python -m pytest`” consideration?

Reply:
@FRidh, wdyt?

Reply:
you can see that the fil is being collected from the examples directory. you want:
```
checkPhase = ''
  pytest tests
'';
```
so that you only bring in examples from the tests directory

Reply:
And, correspondingly, I don't use `pytestCheckHook`?

Reply:
correct to both